pkg/streamingpromql/testdata/ours/selectors.test

```diff
@@ -19,3 +19,36 @@ eval range from 0 to 4m step 1m some_metric
 # If no series are matched, we shouldn't return any results.
 eval range from 0 to 4m step 1m some_nonexistent_metric
   # Should return no results.
+
+# Range query with positive offset (looking backwards).
+# Output T=0 should be looking for sample at T=-1m, hence no results
+# Output T=1m to T=5m should be looking for sample at one minute earlier
+# T=6m should be looking for sample at T=5m, and then fallback to lookback to previous point within 5m lookback window
+eval range from 0 to 6m step 1m some_metric offset 1m
+  some_metric{env="prod", cluster="eu"} _ 0 1 2 3 4 4
+  some_metric{env="prod", cluster="us"} _ 0 2 4 6 8 8
+  some_metric{env="test", cluster="eu"} _ 0 3 6 9 12 12
+  some_metric{env="test", cluster="us"} _ 0 4 8 12 16 16
+
+# Range query with negative offset (looking forwards).
+# Output T=0m to T=3m should be looking for sample at one minute in the future
+# T=5m should be looking for sample at T=4m, and then fallback to lookback to previous point within 5m lookback window
+eval range from 0 to 4m step 1m some_metric offset -1m
+  some_metric{env="prod", cluster="eu"} 1 2 3 4 4
+  some_metric{env="prod", cluster="us"} 2 4 6 8 8
+  some_metric{env="test", cluster="eu"} 3 6 9 12 12
+  some_metric{env="test", cluster="us"} 4 8 12 16 16
+
+# Range query with @ modifier.
+eval range from 0 to 10m step 1m some_metric @ 2m
```

Comment:
what about a testing
`eval range from 0 to 10m step 1m some_metric @ 20m`
and
`eval range from 0 to 10m step 1m some_metric @ -5m`

Reply:
Added in 329094c.